pkg/controller/instance/instance_controller.go

```diff
@@ -548,19 +521,114 @@ func isUpgradePlan(planName string) bool {
 	return planName == v1beta1.DeployPlanName || planName == v1beta1.UpgradePlanName
 }
 
-// getPlanToBeExecuted returns name of the plan that should be executed
-func getPlanToBeExecuted(i *v1beta1.Instance, ov *v1beta1.OperatorVersion) (*string, error) {
+func areWebhooksEnabled() bool {
+	return strings.ToLower(os.Getenv("ENABLE_WEBHOOKS")) == "true"
+}
+
+// getPlanInProgress method returns current plan that is in progress. As long as we don't enforce webhooks
+// (ENABLE_WEBHOOKS=true) we have TWO WAYS of deciding that:
+//
+// 1. WITHOUT the webhook, we have the old logic which iterates over all existing PlanStatuses, searching for
+//    the first one with Status == InProgress. It might happen that multiple plans are in progress (e.g. cleanup and deploy)
+// 2. WITH the webhook, current plan lives in the Spec.PlanExecution.PlanName field
+func getPlanInProgress(i *v1beta1.Instance) *kudov1beta1.PlanStatus {
```

Comment:
uh not a big fan of this ;( I don't like that controller has to check whether some env is set. Will we be able to get rid of this once webhooks are mandatory?

Could we possibly get rid of this? Why is this logic not valid? 🤔 
- get plan in progress based on status
- get plan in progress based on Spec.PlanExecution.PlanName
- if those two are the same, this is your plan
- if Spec.PlanExecution.PlanName is empty, take what is in status, if status does not yield anything, it's new plan triggered via Spec.PlanExecution.PlanName
- if both are empty - try to figure out new plan execution the old way
- if those two are non-empty and differ, take what's in Spec.PlanExecution.PlanName (that way we respect what user gave us - whether he/she was using webhooks or not, if they are not, their bad :) they can if doing not allowed actions interrupt plans in way that's not normally allowed)

 I'll think about this, it would be so nice if this was gone.

Reply:
>  I don't like that controller has to check whether some env is set. Will we be able to get rid of this once webhooks are mandatory?

I don't like it either und we wanted to discuss whether webhooks should become mandatory after this PR lands.

> Why is this logic not valid?

tl;dr: we don't need it. With webhooks, all the logic you need is in the L536 below. There is no ambiguity as to what plan has to be executed: webhook sets it **before** the execution begins and resets it **after** the plan is terminal. We could have sanity checks for error cases like `if those two are non-empty and differ` but that's just it - sanity checks. 

In general, once the webhook becomes mandatory we can get rid of all the _if-WebhooksEnabled-duality_ in the controller code. For now, I'd rather cleanly separate both branches.